fpdf/html.py

```diff
@@ -447,10 +473,12 @@ def _new_paragraph(
         indent=0,
         bullet="",
     ):
+        if bullet and top_margin:
+            raise NotImplementedError(
```

Comment:
This error is aimed at **developpers**, not toward end-users (this is a `NotImplementedError` exception, not a `ValueError` exception).

I got tricked by this at some point, and wanted to keep track of that pitfall somehow.

If you are suggesting that we could fix this down in `TextRegion._render_column_lines()`, I agree, but the impact of this change is too large to be part of this single PR, IMHO.

If you are suggesting another approach, please provide a bit more details so that I can better understand your idea.

Reply:
I don't understand how saying "x will be ignored" while actually throwing an exception can make any sense (no matter the type of exception).

What is currently the actual result when supplying both a bullet and a top margin?

Reply:
The relevant part of the code is [text_region.py L541-548](https://github.com/py-pdf/fpdf2/blob/3b6cd42f9c76c8ef021996ebb3c6b5001535d8d3/fpdf/text_region.py#L541C1-L548C22), where the bullet actually gets rendered.
If you add the paragraph top margin to `pdf.y` right before, and subtract it again after (so the rest of the code doesn't need to special case bullets), then the bullet should end up in the expected place.

Unless I'm missing something very basic here, there's really no need to throw any kind of exception in this situation.

PS.: Also generally speaking: If the only problem is indeed that the bullet gets rendered in the wrong place, then that is of course a bug, and may be annoying, but still no reason to throw an exception in the first place. The users can decide themselfes if they want to accept this result, and don't need any nannying from us.

Reply:
> To be extra clear: with the current code in this MR, this exception will NEVER be raised.

So you're adding code that cannot be executed, and thus is impossible to test. That is not quite my idea of "defensive programming"... :wink:

If you're not willing to fix the underlying problem (which is perfectly fine), just add a comment explaining it (which also explains why that last `_ln()` call is still in there in the "li" section), maybe create a new issue to put it on the to-do list, but otherwise leave it alone.

I'd very strongly recommend against deliberately adding [dead code](https://en.wikipedia.org/wiki/Dead_code), ever, for whatever reason.

Reply:
As I tried to explain patiently, this is not dead code... 😞

But I'm getting tired of this controversy, and I'm glad to give up:
I'm going to remove this condition & exception.